### PR TITLE
OCPBUGS-16629: Increase gateway migration rollout for OVNK-IC

### PR DESCRIPTION
**Pull request by tssurya**
/cc @anuragthehatter @jluhrsen @trozet

### CODE REVIEW

**Comment by jluhrsen** [comment]
/hold
I'm trying to nail down what/where this extra timing got introduced and I'm pushing PRs and running this job to
investigate. if this makes it in, the jobs will pass and it will be harder to figure out. Unless there is some other
way I can do this?

**Comment by tssurya** [comment]
> /hold I'm trying to nail down what/where this extra timing got introduced and I'm pushing PRs and running this job to investigate. if this makes it in, the jobs will pass and it will be harder to figure out. Unless there is some other way I can do this?

I think from the job itself we should be able to measure the time for OVNK components versus K8s rollout perhaps and go with that? We anyways need to do this I guess for IC. But I am OK to wait if you think this will help your investigations...
I'll let you decide when you want to unhold...

**Comment by jluhrsen** [comment]
/hold cancel
/approve
/lgtm

This is going to affect all jobs, not just master. There has definitely been a regression so adding this extra timeout is just ignoring that for now. I have [a bug](https://issues.redhat.com/browse/OCPBUGS-17391) assigned to me that we can use to track this regression. It's possible we will just ignore it, but I need to point out that this is not fixing anything just allowing more time for whatever started taking
longer.

File: ci-operator/step-registry/ovn/local-to-shared-gateway-mode-migration/ovn-local-to-shared-gateway-mode-migration-commands.sh [hunk 1]
```diff
@@ -10,8 +10,10 @@ oc patch Network.operator.openshift.io cluster --type='merge' --patch "{\"spec\"
 
 oc wait co network --for='condition=PROGRESSING=True' --timeout=60s
 # Wait until the ovn-kubernetes pods are restarted
-timeout 360s oc rollout status ds/ovnkube-node -n openshift-ovn-kubernetes
-timeout 360s oc rollout status ds/ovnkube-master -n openshift-ovn-kubernetes
+# FIXME: Increasing timeout to 15minutes for OVNK IC deployments (original value was 360seconds)
+# See https://issues.redhat.com/browse/OCPBUGS-16629 for details
+timeout 900s oc rollout status ds/ovnkube-node -n openshift-ovn-kubernetes
+timeout 900s oc rollout status ds/ovnkube-master -n openshift-ovn-kubernetes
```
Comment (tssurya) [inline ci-operator/step-registry/ovn/local-to-shared-gateway-mode-migration/ovn-local-to-shared-gateway-mode-migration-commands.sh:16]:
```
+ timeout 900s oc rollout status ds/ovnkube-master -n openshift-ovn-kubernetes
Error from server (NotFound): daemonsets.apps "ovnkube-master" not found
```
this is not `ovnkube-master` any more in IC world, its `ovnkube-control-plane`

Comment (tssurya) [reply ci-operator/step-registry/ovn/local-to-shared-gateway-mode-migration/ovn-local-to-shared-gateway-mode-migration-commands.sh:16]: also its not daemonset anymore,
its deployment

**Comment by tssurya** [comment]
/pj-rehearse pull-ci-openshift-ovn-kubernetes-release-4.14-e2e-aws-ovn-shared-to-local-gateway-mode-migration
/pj-rehearse pull-ci-openshift-ovn-kubernetes-release-4.15-e2e-aws-ovn-local-to-shared-gateway-mode-migration
/pj-rehearse pull-ci-openshift-ovn-kubernetes-release-4.13-e2e-aws-ovn-local-to-shared-gateway-mode-migration

**Comment by tssurya** [comment]
/pj-rehearse pull-ci-openshift-ovn-kubernetes-master-e2e-aws-ovn-local-to-shared-gateway-mode-migration
/pj-rehearse pull-ci-openshift-ovn-kubernetes-master-e2e-aws-ovn-shared-to-local-gateway-mode-migration

**Comment by tssurya** [comment]
/pj-rehearse pull-ci-openshift-ovn-kubernetes-master-e2e-aws-ovn-shared-to-local-gateway-mode-migration

**Comment by tssurya** [comment]
/pj-rehearse pull-ci-openshift-ovn-kubernetes-release-4.13-e2e-aws-ovn-local-to-shared-gateway-mode-migration

**Comment by tssurya** [comment]
/pj-rehearse pull-ci-openshift-ovn-kubernetes-release-4.15-e2e-aws-ovn-shared-to-local-gateway-mode-migration

**Comment by tssurya** [comment]
/hold cancel
fixed the issue we were facing

**Comment by tssurya** [comment]
/pj-rehearse pull-ci-openshift-ovn-kubernetes-release-4.14-e2e-aws-ovn-shared-to-local-gateway-mode-migration

**Comment by tssurya** [comment]
/pj-rehearse pull-ci-openshift-ovn-kubernetes-release-4.14-e2e-aws-ovn-local-to-shared-gateway-mode-migration

**Comment by tssurya** [comment]
IC:
https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_release/42165/rehearse-42165-pull-ci-openshift-ovn-kubernetes-master-e2e-aws-ovn-local-to-shared-gateway-mode-migration/1693586734966640640/artifacts/e2e-aws-ovn-local-to-shared-gateway-mode-migration/ovn-local-to-shared-gateway-mode-migration/build-log.txt
===
https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_release/42165/rehearse-42165-pull-ci-openshift-ovn-kubernetes-master-e2e-aws-ovn-shared-to-local-gateway-mode-migration/1693587184717664256/artifacts/e2e-aws-ovn-shared-to-local-gateway-mode-migration/ovn-shared-to-local-gateway-mode-migration/build-log.txt

nonIC:
https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_release/42165/rehearse-42165-pull-ci-openshift-ovn-kubernetes-release-4.13-e2e-aws-ovn-local-to-shared-gateway-mode-migration/1693587238429921280/artifacts/e2e-aws-ovn-local-to-shared-gateway-mode-migration/ovn-local-to-shared-gateway-mode-migration/build-log.txt
===

**Pull request merged**